students/jack_anderson/mailroom/mailroom.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+"""
+Jack Anderson
+02/14/2020
+UW PY210
+Mailroom assignment part 1
+"""
+
+
+# The list of donors and their amounts donated
+donors_list = [
+    ['Bubbles Trailer',[1500.23, 2523.12, 3012.89]],
+    ['Julien Park',[2520.92, 8623.12]],
+    ['Ricky Boys',[7042.76, 3845.56, 5123.25]],
+    ['Jack Anderson',[1044, 2232, 4123.56]],
+    ['Lacey Coffin Greene',[7500.32, 6625, 1305, 3400.78]]
+    ]
+
+def donor_names():
+    # Return a list of donor names
+    names = []
+    for donors in donors_list:
+        names.append(donors[0])
+    return names
+
+def donor_details(x):
+    """
+    Action to summarize the details of a donor in the list
+    :param x: The donors index number in the donor list
+    :return: Name of donor, total amount donated, number of donations donor made, avg donation of donor
+    """
+    donor = x
+    name = donor[0]
+    donations = donor[1]
+    num_donations = len(donor[1])
+    totals = total_donated(donations)
+    avg_donation = int(totals) / int(num_donations)
+    return(name, totals, num_donations, avg_donation)
+
+
+def name_check(x):
+    """
+    Action to check if name exists in list and add name if not exists
+    :param x: Name of donor
+    :return: The index number of Name parameter
+    """
+    count = len(donor_names())
+    current = donor_names()
+    start = 0
+    while start < count:
+        if x == current[start]:
+            return(start)
+        start = start + 1
+    new_donor = [x,[]]
```

Comment:
this can be simplified with a for loop and is preferred:
```
for donor in donors_list:
    if donor[0] == x:
      return donor
  else:
    new_donor = [x, []]
    donors_list.append(new_donor)
    return new_donor
```